### PR TITLE
Passed timeout to info.options

### PR DESCRIPTION
**Pull request by anatolybolshakov**
It seems that timeout is not being passed to request info - fixed it and passed it to request info also.

### CODE REVIEW

File: lib/HttpClient.ts [hunk 1]
```diff
@@ -418,6 +418,7 @@ export class HttpClient implements ifm.IHttpClient {
         info.options.port = info.parsedUrl.port ? parseInt(info.parsedUrl.port) : defaultPort;
         info.options.path = (info.parsedUrl.pathname || '') + (info.parsedUrl.search || '');
         info.options.method = method;
+        info.options.timeout = this.requestOptions && this.requestOptions.socketTimeout;
```
Comment (damccorm) [inline lib/HttpClient.ts:421]: I think we want to also check (and set) `this._socketTimeout`. I think it should look something like

```
info.options.timeout = (this.requestOptions && this.requestOptions.socketTimeout) || this._socketTimeout;
this._socketTimeout = info.options.timeout;
```

**Review by damccorm — APPROVED** [review]

**Comment by damccorm** [comment]
Test failures here look unrelated to the change, I think httpbin is having some issues https://github.com/microsoft/typed-rest-client/issues/236

**Comment by damccorm** [comment]
I'm comfortable merging without tests passing here because of that - @anatolybolshakov I realized you didn't have write permissions, I just added the team so you should have them now

**Comment by anatolybolshakov** [comment]
> I'm comfortable merging without tests passing here because of that - @anatolybolshakov I realized you didn't have write permissions, I just added the team so you should have them now

Ok, thanks! So I'm merging it currently

**Comment by damccorm** [comment]
Published!

**Pull request merged**